rotkehlchen/assets/utils.py

```diff
@@ -238,6 +261,14 @@ def get_or_create_evm_token(
 
                 with userdb.user_write() as cursor:
                     userdb.add_asset_identifiers(cursor, [evm_token.identifier])
+            # at this point the asset is in the database
+            if is_spam_token:
+                with userdb.user_write() as write_cursor:
+                    if write_cursor.execute(
```

Comment:
Why do we need this if?

1. Has it not just been added to the DB? So either it happens at addition (at which case this is not needed) or this is always true. Perhaps we need a more well defined, "Add new token to the user and global DB" function.
2. I see that `add_to_ignored_assets` does not do INSERT OR IGNORE in which case this if is needed. Can't we change it? This extra logic is silly and not sure if we have it elsewhere too.

Reply:
I have moved it to only addition and also changed the insert to insert or ignore